cdap-app-templates/cdap-program-report/cdap-program-report-base/src/main/scala/co/cask/cdap/report/ReportGenerationHelper.scala

```diff
@@ -95,6 +96,15 @@ object ReportGenerationHelper {
   @throws(classOf[IOException])
   def generateReport(sql: SQLContext, request: ReportGenerationRequest, inputURIs: java.util.List[String],
                      reportIdDir: Location): Unit = {
+    // if there is no input avro file, save 0 as the number of records in the report and mark the report generation
+    // as completed
+    if (Option(inputURIs).isEmpty || inputURIs.isEmpty) {
+      // Save an enmpty report summary in the _SUMMARY file in the given directory
+      writeToFile(GSON.toJson(ReportSummary.getEmptyReportSummary(request.getStart, request.getEnd)),
```

Comment:
why do you need to write this ? what will happen if you don't write this empty report summary ? is it because we want to return the information we got as part of query and skip all the additions ? in that case we need to have requestSummary class and report summary should extend that class, and in this case when there is no data, we just write the request summary , while there are results, write report summary

Reply:
could handle serialization using adapter if formatting is an issue.